Modules/Sources/Storage/GRDB/Migrations/V001InitialSchema.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import GRDB
+
+struct V001InitialSchema {
+    // This migration is under development and not released yet.
+    // It's still open for modification, until we ship.
+    // TODO: Mark this as final when we enable the pointOfSaleLocalCatalogi1 feature flag
+
+    static func migrate(_ db: Database) throws {
+        try createProductTable(db)
+        try createProductAttributeTable(db)
+        try createProductImageTable(db)
+        try createProductVariationTable(db)
+        try createProductVariationAttributeTable(db)
+        try createProductVariationImageTable(db)
+    }
+
+    static func createProductTable(_ db: Database) throws {
+        // Note that it's best to use strings for names in the database
+        // not derive them from a Swift class.
+        // https://swiftpackageindex.com/groue/grdb.swift/v7.6.1/documentation/grdb/migrations#Good-Practices-for-Defining-Migrations
+        try db.create(table: "product") { productTable in
+            productTable.primaryKey(["siteID", "productID"])
+
+            productTable.column("siteID", .integer).notNull()
+            productTable.column("productID", .integer).notNull()
+
+            productTable.column("name", .text).notNull()
+            productTable.column("productTypeKey", .text).notNull()
+
+            productTable.column("fullDescription", .text)
+            productTable.column("shortDescription", .text)
+
+            productTable.column("sku", .text)
+            productTable.column("globalUniqueID", .text)
+            productTable.column("price", .text).notNull()
+
+            productTable.column("downloadable", .boolean).notNull()
+
+            productTable.column("parentID", .integer).notNull()
+        }
+    }
+
+    private static func createProductAttributeTable(_ db: Database) throws {
+        try db.create(table: "productAttribute") { productAttributeTable in
+            productAttributeTable.primaryKey(["siteID", "attributeID"])
+
+            productAttributeTable.column("siteID", .integer).notNull()
+            productAttributeTable.column("attributeID", .integer).notNull()
+            productAttributeTable.column("productID", .integer).notNull()
+
+            productAttributeTable.column("name", .text).notNull()
+            productAttributeTable.column("position", .integer).notNull()
+            productAttributeTable.column("visible", .boolean).notNull()
+            productAttributeTable.column("variation", .boolean).notNull()
+            productAttributeTable.column("options", .jsonText).notNull()
+
+            productAttributeTable.foreignKey(["siteID", "productID"], references: "product")
```

Comment:
I explored using `belongsTo` instead of `foreignKey` here. It's a bit nicer, because it automatically makes the required foreign key fields on the table... but it's a bit odd with composite keys.

I think it would create `productProductID`, `productSiteID` columns in this case.

We could make a top level `Site` entity and then `Products`/`Variations` could all belong to a single site. That would let us use the `belongsTo` approach. WDYT?

Reply:
OK, I'll give that a go. [We can set cascade deletes with `foreignKey` as well](https://swiftpackageindex.com/groue/grdb.swift/v7.6.1/documentation/grdb/columndefinition/references(_:column:ondelete:onupdate:deferred:)). `belongsTo` is a nice shorthand though.